core/src/main/scala/cats/syntax/all.scala

```diff
@@ -48,3 +48,6 @@ trait AllSyntax
     with ValidatedSyntax
     with VectorSyntax
     with WriterSyntax
+
+trait AllSyntaxBinCompat0
```

Comment:
One more small thing, if we make this one abstract class (and maybe even have it extends AllSyntax) we can add more traits to it going forward.

Reply:
How about a `abstract class AllSyntaxBinCompat extends AllSyntax with AllSyntaxBinCompat0`?

Reply:
then what's the purpose of AllSyntaxBinCompat0 then? just as a versioning scheme?

Reply:
A downstream consumer can't mix in the abstract class but they could mix in `AllSyntax` and `AllSyntaxBinCompat0` to get roughly the same effect. Without `AllSyntaxBinCompat0`, but then you'd have to remember that `AllSyntax` is missing `UnorderedTraverse` syntax. So it's versioning scheme to facilitate downstream use.

I'm not sure if it's worth it though. It's just an idea.

Reply:
I see. Makes sense to me